Code/Framework/Tests/InstanceDataHierarchy.cpp

```diff
@@ -1314,4 +1410,243 @@ namespace UnitTest
+        while (!nodeStack.empty())
+        {
+            InstanceDataNode* node = nodeStack.top();
+            nodeStack.pop();
+            if (node->GetClassMetadata()->m_typeId == AZ::AzTypeInfo<GroupTestComponent>::Uuid())
+            {
+                componentNode1 = node;
+                break;
+            }
+            for (InstanceDataNode& child : node->GetChildren())
+            {
+                nodeStack.push(&child);
+            }
+        }
+        // Iterating through the children in the instance data hierarchy to verify their properties
+        ASSERT_TRUE(componentNode1 != nullptr);
+        for (auto child : componentNode1->GetChildren())
+        {
+            AZStd::string childName(child.GetElementMetadata()->m_name);
+            if (childName.compare("GroupFloat") == 0)
+            {
+                // False for any child node with serializable data
+                ASSERT_FALSE(child.GetElementEditMetadata()->IsClassElement());
+                // Child node should never be a ClassElement::Group, unless it is the root node of a ToggleGroup 
+                ASSERT_NE(child.GetElementEditMetadata()->m_elementId, AZ::Edit::ClassElements::Group);
+                // Ensuring that this node was assigned to the appropriate group
+                ASSERT_EQ(child.GetGroupElementMetadata()->m_description, "Normal Group");
+                // Ensuring that this node has the correct parent
+                ASSERT_EQ(child.GetParent()->GetClassMetadata()->m_name, "GroupTestComponent"); 
+            }
+        }
+    }
+
+    TEST_F(InstanceDataHierarchyGroupTestFixture, TestToggleGroups)
+    {
+        using namespace AzToolsFramework;
+
+        // Setting up the data node hierarchy
+        AZ::SerializeContext serializeContext;
+        serializeContext.CreateEditContext();
+        Entity::Reflect(&serializeContext);
+        GroupTestComponent::Reflect(&serializeContext);
+
+        AZStd::unique_ptr<AZ::Entity> testEntity1(new AZ::Entity());
+        testEntity1->CreateComponent<GroupTestComponent>();
+
+        InstanceDataHierarchy instanceDataHierarchy;
+        instanceDataHierarchy.AddRootInstance(testEntity1.get());
+        instanceDataHierarchy.Build(&serializeContext, 0);
+
+        // Adding the nodes to a node stack
+        auto rootNode = instanceDataHierarchy.GetRootNode();
+        AZStd::stack<InstanceDataNode*> nodeStack;
+        nodeStack.push(rootNode);
+        InstanceDataNode* componentNode1 = nullptr;
+        while (!nodeStack.empty())
+        {
+            InstanceDataNode* node = nodeStack.top();
+            nodeStack.pop();
+            if (node->GetClassMetadata()->m_typeId == AZ::AzTypeInfo<GroupTestComponent>::Uuid())
+            {
+                componentNode1 = node;
+                break;
+            }
+            for (InstanceDataNode& child : node->GetChildren())
+            {
+                nodeStack.push(&child);
+            }
+        }
```

Comment:
suggestion: This code looks almost identical to the code in the other test - it might be worth making a helper function that builds this nodeStack for you (it's fine to have test setup code moved into separate functions so long as it doesn't dramatically impact the readability of the test). Might be worth refactoring this a little bit to remove the duplication

Reply:
Sounds good, I moved this bit of code to a helper method in the test fixture.